python/tank/bootstrap/manager.py

```diff
@@ -236,32 +244,153 @@ def bootstrap_engine(self, engine_name, entity=None):
+        to business logic set in the associated resolver class and based
+        on this launch a configuration. This may involve downloading new
+        apps from the toolkit app store and installing files on disk.
+
+        3 callback functions can be provided.
+
+        A callback function that reports back on the bootstrap progress
+        with the following signature::
+
+            progress_callback(progress_value, message, current_index, maximum_index)
+
+        where:
+        - ``progress_value`` is the current progress value, a float number ranging from 0.0 to 1.0
+                             representing the percentage of work completed.
+        - ``message`` is the progress message string to report.
+        - ``current_index`` is an optional current item number being looped over.
+                            This integer number is relative to ``maximum_index``.
+                            Its value is ``None`` when not provided.
+        - ``maximum_index`` is an optional maximum item number being looped over.
+                            This integer number leads to completion of the current progress step.
+                            Its value is ``None`` when not provided.
+
+        A callback function that handles cleanup after successful completion of the bootstrap
+        with the following signature::
+
+            completed_callback(engine)
+
+        where:
+        - ``engine``is the launched :class:`sgtk.platform.Engine` instance.
+
+        A callback function that handles cleanup after failed completion of the bootstrap
+        with the following signature::
+
+            failed_callback(phase, exception)
+
+        where:
+        - ``phase`` is the bootstrap phase that raised the exception,
+                    ``ToolkitManager.TOOLKIT_BOOTSTRAP_PHASE`` or ``ToolkitManager.ENGINE_STARTUP_PHASE``.
+                    Using this phase, the callback can decide if the toolkit core needs
+                    to be re-imported to ensure usage of a swapped in version.
+        - ``exception`` is the python exception raised while bootstrapping.
+
+        Please note that the API version of the tk instance that hosts
+        the engine may not be the same as the API version that was
+        executed during the bootstrap.
+
+        :param engine_name: Name of engine to launch (e.g. ``tk-nuke``).
+        :param entity: Shotgun entity to launch engine for.
+        :param progress_callback: Callback function that reports back on the toolkit and engine bootstrap progress.
+        :param completed_callback: Callback function that handles cleanup after successful completion of the bootstrap.
+        :param failed_callback: Callback function that handles cleanup after failed completion of the bootstrap.
+        """
+
+        log.info("Bootstrapping engine %s for entity %s." % (engine_name, entity))
+
+        try:
+            import threading
```

Comment:
I am still confused by this check - trying to import threading to probe if qt is available?

I can import threading fine on a non-qt setup:

```
mannes-mbp:developer manne$ python
Python 2.5.6 (r256:88840, Apr 21 2016, 09:13:10) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import PySide
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named PySide
>>> import threading
```